frontend/src/lib/hooks/useSession.tsx

```diff
@@ -0,0 +1,23 @@
+import { useAsync, useSessionStorage } from "react-use";
+
+import { User, whoAmI } from "../api/auth";
+
+export const USER_SESSION_KEY = "oab.lib.hooks.user";
+
+export const useSession = (): { user: User | null } => {
```

Comment:
I think I prefer all I/O to be handled by action/loaders to keep our pages as clean (and sync) as possible, since loaders are just (async) functions, using a hook this way won't work.